CedarJavaFFI/src/interface.rs

```diff
@@ -151,6 +176,123 @@ pub fn json_parse_entity_uid(input: &str) -> InterfaceResult {
     }
 }
 
+#[jni_fn("com.cedarpolicy.value.EntityIdentifier")]
+pub fn getEntityIdentifierRepr<'a>(mut env: JNIEnv<'a>, _: JClass, obj: JObject<'a>) -> jvalue {
+    match get_entity_identifier_repr_internal(&mut env, obj) {
+        Ok(v) => v.as_jni(),
+        Err(e) => jni_failed(&mut env, e.as_ref()),
+    }
+}
+
+fn get_entity_identifier_repr_internal<'a>(
+    env: &mut JNIEnv<'a>,
+    obj: JObject<'a>,
+) -> Result<JValueOwned<'a>> {
```

Comment:
for my own edification, what is `JValueOwned` vs `JValueGen` and why do some of these helpers return one vs the other?

Reply:
`JValueOwned` is a typedef for `JValueGen`, which can represent objects both owned and borrowed